src/server/helpers/entityRouteUtils.js

```diff
@@ -94,3 +96,58 @@ export function entityEditorMarkup(
 
 	return markup;
 }
+
+
+/**
+ * Callback for any transformations to the request body
+ * @callback transformCallback
+ * @param {object} body - request body
+ */
+
+/**
+ * Handler for create or edit actions
+ * @callback transformCallback
+ * @param {object} req - request object
+ * @param {object} res - response object
+ */
+
+/**
+ * Additional callback to pass data to createEntity
+ * @callback createCallback
+ * @param {object} req - request object
+ * @param {object} res - response object
+ */
+
+/**
+ * Makes a middleware handler for create or edit actions on entities.
+ * @param {string} entityType - entity type
+ * @param {transformCallback} transformNewForm - callback
+ * @param {(string|string[])} propertiesToPick - props from transformed
+ * request body to pick.
+ * @param {createCallback} createCallback - a callback that returns any
+ * additional data for createEntity.
+ * @returns {createOrEditHandler} createOrEditHandler - middleware handler
+ */
+export function makeEntityCreateOrEditHandler(
+	entityType: string,
+	transformNewForm: Function,
+	propertiesToPick: string | string[],
+	createCallback?: (req: Object, res: Object) => any =
+	(req, res) => null) {
+	const entityName = _.capitalize(entityType);
+	return function createOrEditHandler(
+		req: express.request,
+		res: express.response) {
+		const validate = getValidator(entityType);
+		if (!validate(req.body)) {
+			const err = new error.FormSubmissionError();
+			error.sendErrorAsJSON(res, err);
+		}
+
+		req.body = transformNewForm(req.body);
+		return entityRoutes.createEntity(
```

Comment:
Looks like this is only creating, and not allowing for editing.

Reply:
Oops! Will fix.